src/app/dwg/qgsdwgimportdialog.cpp

```diff
@@ -535,3 +544,25 @@ void QgsDwgImportDialog::showHelp()
 {
   QgsHelp::openHelp( QStringLiteral( "managing_data_source/opening_data.html#importing-a-dxf-or-dwg-file" ) );
 }
+
+void QgsDwgImportDialog::layers_clicked( QTableWidgetItem *item )
+{
+  if ( ! item )
+    return;
+
+  if ( item->column() != static_cast<int>( ColumnIndex::Name ) )
+    item = mLayers->item( item->row(), static_cast<int>( ColumnIndex::Name ) );
+
+  if ( ! item )
+    return;
+
+  const QgsTemporaryCursorOverride waitCursor( Qt::WaitCursor );
+
+  QString layerName = item->text();
+
+  QgsLayerTreeGroup layerGroup;
+  createGroup( &layerGroup, layerName, QStringList( layerName ), true );
```

Comment:
If you follow the chain of function calls this is adding layers to the current project -- not what we want at all for a preview! You'll need to refactor out the layer creation and store the preview layer in a unique_ptr in the dialog only.

Reply:
Thank you, I did not see it. I added a new boolean parameter to createGroup and layer functions `addToProject`.

Reply:
If add to project is false then the layer is leaked. 

I think this needs more extensive refactoring -- the actual layer creation and styling code should be made independent of the layer group handling. Then the preview functionality should just call the layer creator function, store the result in a member unique ptr and set the canvas layer accordingly. There's no need for the preview functionality to handle layer tree related logic at all.

Reply:
Make sense thanks. But the layers to preview are multiple, one dxf layer correspond to multiple qgis layers, so I put them on a list and delete it in the destructor

Reply:
Looking good, thanks!